api/src/controllers/fetcher.ts

```diff
@@ -0,0 +1,100 @@
+import { PrismicDocument, Query } from "@prismicio/types";
+import * as prismic from "@prismicio/client";
+import { isString } from "../helpers";
+import { ArticlePrismicDocument, ContentType } from "../types";
+
+const graphQuery = `{
+    articles {
+      title
+      format {
+        title
+      }
+      promo
+      contributors {
+        ...contributorsFields
+        role {
+          title
+        }
+        contributor {
+          ... on people {
+            name
+          }
+          ... on organisations {
+            name
+          }
+        }
+      }
+    }
+  }`.replace(/\n(\s+)/g, "\n");
+
+type GetServerSidePropsPrismicClient = {
+  type: "GetServerSidePropsPrismicClient";
+  client: prismic.Client;
+};
+
+type GetByTypeParams = Parameters<
+  GetServerSidePropsPrismicClient["client"]["getByType"]
+>[1];
+
+const contentTypes: ContentType[] = ["articles", "webcomics"];
+export const articlesFetcher = fetcher<ArticlePrismicDocument>(contentTypes);
+
+export function fetcher<Document extends PrismicDocument>(
```

Comment:
Ah I guess this whole thing is copied from wc.org - it's rather confusing, I wonder if we can just use the prismic client as-is for our needs in this app?